packages/app/src/runner/index.ts

```diff
@@ -244,7 +260,37 @@ function runSpecE2E (spec: BaseSpec) {
  * This only needs to happen once, prior to running the first spec.
  */
 async function initialize () {
+  isTorndown = false
+
   await injectBundle()
+
+  if (isTorndown) {
```

Comment:
This might look weird, "how can `isTorndown` be `true` here? The edge case is this:

1. Call `injectBundle`
2. Before it finishes, [onBeforeUnmount](https://github.com/cypress-io/cypress/pull/19372/files#diff-d37cab2179f306cb9a684a1a871c60981f47c386088215280a72bdf79a26df13R34) is called, eg when navigating away from or refreshing the page. The teardown function sets `isTorndown` to `true`
3. In this case, we do not want to continue - just return early.